src/main/java/cs2103/v15_1j/jimjim/searcher/KeywordFilter.java

```diff
@@ -21,20 +21,20 @@ public KeywordFilter(List<String> keywords) {
 
     @Override
     public boolean check(TaskEvent taskEvent) {
+    	if (keywords.size() == 1 && keywords.get(0).length() == 1) { // Single char
+    		char keyword = keywords.get(0).charAt(0);
+    		if (Character.toLowerCase(keyword) == Character.toLowerCase(taskEvent.getName().charAt(0))) {
+    			return true;
+    		}
+    	}
     	List<String> wordsInName = Arrays.asList(taskEvent.getName().split(" "));
-    	for (String keyword : keywords) {
-    		boolean found = false;
-    		for (String word : wordsInName) {
-    			if (word.equalsIgnoreCase(keyword) || 
-    				Character.toLowerCase(word.charAt(0)) == Character.toLowerCase(keyword.charAt(0))) {
-    				found = true;
-    				break;
+    	for (String word : wordsInName) {
+    		for (String keyword : keywords) {
+    			if (word.equalsIgnoreCase(keyword)) {
+    				return true;
```

Comment:
That's wrong, I don't want to return false if the first keyword fails. I want to check if the other keywords also match